src/broker/broker.c

```diff
@@ -735,6 +738,11 @@ static int checkdir (const char *name, const char *path)
         log_err ("cannot stat %s %s", name, path);
         return -1;
     }
+    if (sb.st_uid != getuid ()) {
```

Comment:
Should `sb.st_uid == 0` also be allowed? (Actually, I don't know the reason for the directory ownership requirement, instead of just write access. Is it for security purposes (to ensure an existing content store could not have been modified by another user), or just to make it simpler to check for write access?)

If this requirement stays, then we might want to update the documentation of `statedir` to include something like: "If set, `statedir` must exist and be owned by the instance owner"

Reply:
My thought was that the tests for u=rwx does not have the desired effect if "u" is not the instance owner.  So we could just make the test more sophisticated, looking for write access to the directory by `getuid()`?   That seemed a bit involved without a use case in mind.  Does that make sense?

Reply:
> My thought was that the tests for u=rwx does not have the desired effect if "u" is not the instance owner.

Yes, that part was clear. My thought is that it wasn't clear you can't specify a `statedir` to which you have write access but are not the owner. A scenario that is unlikely but was the first thing I tried (`statedir=/tmp`). It also might be the case that sysadmins might prefer a directory ownership of `root` for a statedir location for the system instance.

However, I agree there probably isn't a real use case at this time, and if we need to fix it the code already exists in `flux-security` to check a directory for secure permissions exhaustively, so we can just bring that code in when necessary.

My main suggestion was just to add documentation of the requirement at this point.

Reply:
Got it.  OK, I'll make the documentation change.